//
// Aeca
//
// Copyright (c) 2023-2024 Aeca, Inc.
//

// clang-format off
// protoc --cpp_out=. fts_analysis_pipeline.proto
// protoc --grpc_out=. --plugin=protoc-gen-grpc=`which grpc_cpp_plugin` fts_analysis_pipeline.proto
// clang-format on

syntax = "proto3";

package aeca.rpc.db.fts;

option cc_enable_arenas = true;

service FTSAnalysisPipelineService {
  rpc execute(PipelineExecutionRequest) returns (PipelineExecutionResponse) {}
  rpc execute_adhoc(AdhocPipelineExecutionRequest)
      returns (AdhocPipelineExecutionResponse) {}
}

message ProfileInfo { uint64 duration_us = 1; }

message PipelineExecutionRequest {
  string collection_name = 1;
  string index_name = 2;
  repeated string field_names = 3;
  string query = 4;
}

message PipelineExecutionResponse {
  int32 status = 1;
  string result = 2;
  ProfileInfo profile = 3;
}

message AdhocPipelineExecutionRequest {
  string pipeline_def = 1;
  repeated string field_names = 2;
  string query = 3;
}

message AdhocPipelineExecutionResponse {
  int32 status = 1;
  string result = 2;
  ProfileInfo profile = 3;
}
